### PR TITLE
Rdm 3916 case filter new contract

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-3916

### Change description ###

CreateCaseFilters component now accepts data and events from consumer component. Also, errors are handled by consumer.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by adnanakgun** [comment]
Checks will fail because https://github.com/hmcts/ccd-case-ui-toolkit/pull/236 needs to be merged first.

File: src/app/cases/creator/case-creator.component.html [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+        </li>
+    </ul>
+    </div>
+    <ccd-callback-errors
+        [triggerTextContinue]="triggerTextStart"
+        [triggerTextIgnore]="triggerTextIgnoreWarnings"
+        [callbackErrorsSubject]="callbackErrorsSubject"
+        (callbackErrorsContext)="callbackErrorsNotify($event)">
+    </ccd-callback-errors>
+    <ccd-create-case-filters
```
Comment (andrewfolga) [inline src/app/cases/creator/case-creator.component.html]: I think I get the error as input now 👍 😄 Looks good

File: src/app/cases/creator/case-creator.component.html [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+        [jurisdictions]="jurisdictions"
+        [error]="error"
+        [triggerText]="triggerText"
```
Comment (andrewfolga) [inline src/app/cases/creator/case-creator.component.html]: Is it possible to change the name here to startButtonText?

**Comment by andrewfolga** [comment]
Looks better now 👍

File: src/app/cases/creator/case-creator.component.ts [hunk 3]
```diff
@@ -1,18 +1,75 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
-import { Profile } from '@hmcts/ccd-case-ui-toolkit';
+import { Subject } from 'rxjs';
+import { HttpError, Jurisdiction, CallbackErrorsContext, AlertService } from '@hmcts/ccd-case-ui-toolkit/';
+import { Router } from '@angular/router';
+import { CreateCaseFiltersSelection } from '@hmcts/ccd-case-ui-toolkit/dist/shared/components';
```
Comment (andrewfolga) [inline src/app/cases/creator/case-creator.component.ts]: Can we follow `import { HttpError, Jurisdiction, CallbackErrorsContext, AlertService } from '@hmcts/ccd-case-ui-toolkit/';` please

**Pull request merged**